### PR TITLE
[#53] [Chore] Setup coverage report

### LINKED ISSUE #53: [Chore] Setup coverage report
## Why

We will have more insight for the code reviewer if we can show the current code coverage for the iOS project and KMM library in  a PR

## Who Benefits?

Developer

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/53

## What happened 👀

- Setup Danger report for the iOS project
- Setup Danger report for the Shared project

## Insight 📝

As far as I researched, there is no way to integrate Jacoco into the KMM project so we are unable to create a code coverage report for the Shared project

## Proof Of Work 📹

CleanShot 2023-04-14 at 16 09 57

CleanShot 2023-04-14 at 16 09 49

### CODE REVIEW

**Comment by markgravity** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  🙁 Detekt violations found: **5**.
Please fix them to proceed. We have zero-warning policy
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:construction: PR is marked with Work in Progress (WIP)</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **/home/runner/work/kmm-ic/kmm-ic/shared/src/androidMain/kotlin/co/nimblehq/mark/kmmic/Platform.kt#L3** - 
**Detekt**: The file name 'Platform' does not match the name of the single top-level declaration 'AndroidPlatform'.
**Rule**: detekt.naming.MatchingDeclarationName
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **/home/runner/work/kmm-ic/kmm-ic/shared/src/androidTest/kotlin/co/nimblehq/mark/kmmic/androidTest.kt#L6** - 
**Detekt**: The file name 'androidTest' does not match the name of the single top-level declaration 'AndroidGreetingTest'.
**Rule**: detekt.naming.MatchingDeclarationName
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **/home/runner/work/kmm-ic/kmm-ic/shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/commonTest.kt#L6** - 
**Detekt**: The file name 'commonTest' does not match the name of the single top-level declaration 'CommonGreetingTest'.
**Rule**: detekt.naming.MatchingDeclarationName
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **/home/runner/work/kmm-ic/kmm-ic/shared/src/iosMain/kotlin/co/nimblehq/mark/kmmic/Platform.kt#L5** - 
**Detekt**: The file name 'Platform' does not match the name of the single top-level declaration 'IOSPlatform'.
**Rule**: detekt.naming.MatchingDeclarationName
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **/home/runner/work/kmm-ic/kmm-ic/shared/src/iosTest/kotlin/co/nimblehq/mark/kmmic/iosTest.kt#L6** - 
**Detekt**: The file name 'iosTest' does not match the name of the single top-level declaration 'IosGreetingTest'.
**Rule**: detekt.naming.MatchingDeclarationName
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems">Danger Kotlin</a> against 4fef0152003f5a32fd5d02a7d2afe3cbad5c3f0b
</p>

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -4,6 +4,14 @@ on:
   pull_request:
     types: [opened, reopened, edited, synchronize]
 
+permissions:
+  actions: write
```
Comment (phongvhd93) [inline .github/workflows/test.yml:8]: Would you let me know why we need these permission

File: ios/Dangerfile
Comment (phongvhd93) [inline ios/Dangerfile:1]:
```
# SwiftFormat
swiftformat.binary_path = './Pods/SwiftFormat/CommandLineTool/swiftformat'
swiftformat.exclude = %w(Pods/**  **/*generated.swift)
swiftformat.check_format
```

Please consider to exclude `fastlane` folder

File: .github/workflows/test.yml (hunk 1 above)
Comment (markgravity) [reply .github/workflows/test.yml:8]: I added these permissions to resolve this issue from Danger-Kotlin
```
Resource Not Available to Integration
```
Following the [solution](https://github.com/danger/danger-js/issues/1031#issuecomment-1017832635)

File: ios/Dangerfile
Comment (markgravity) [reply ios/Dangerfile:1]: @phongvhd93 updated at [3f6a4e8](https://github.com/markgravity/kmm-ic/pull/59/commits/3f6a4e8794a404b620c3284f18729f52a2d51890)

**Review by phongvhd93 — APPROVED** [review]

**Review by suho — APPROVED** [review]

**Pull request merged**